README.md

```diff
@@ -72,6 +72,27 @@ populated if you simply run `go build`. You have to pass in command
 line flags as defined in `Makefile.INCLUDE` (see `${BUILDFLAGS}`) to
 do that.
 
+## Release checklist
+
+To cut a new release of Prometheus, follow this checklist:
+
+1. Consider updating all or some of the vendored dependencies before the
+   release. Make sure that https://github.com/prometheus/client_golang is
+   vendored at a release version.
+1. Create a PR that updates the `CHANGELOG.md` along with the `version/VERSION` file and get it merged.
+1. Run `make tag` to tag the new release and push it to GitHub.
+1. Head to https://github.com/prometheus/prometheus/releases and create a new release for the pushed tag.
+1. Attach the relevant changelog excerpt to the release on GitHub.
+1. Build `linux-amd64`, `linux-386`, and `darwin-amd64` binaries via `make tarball` and attach them to the release.
+1. Update the `stable` branch to point to the new release (to tag the latest release image on https://registry.hub.docker.com/u/prom/prometheus/):
+
+   ```bash
+   git branch -f stable <new-release-tag>
+   git push origin stable
+   ```
+
+1. Announce the release on the mailing list and on Twitter.
```

Comment:
The readme's intention is to get users started with Prometheus as fast as
possible. Cutting releases is not relevant to 99.99% of them.

This should be a wiki page. Especially as we cannot keep growing the readme
for all development related knowledge.

On Mon, Jul 27, 2015, 12:59 PM Björn Rabenstein <notifications@github.com>
wrote:

> In README.md
> <https://github.com/prometheus/prometheus/pull/930#discussion_r35524793>:
>
> > +1. Consider updating all or some of the vendored dependencies before the
> > +   release. Make sure that https://github.com/prometheus/client_golang is
> > +   vendored at a release version.
> > +1. Create a PR that updates the `CHANGELOG.md` along with the `version/VERSION` file and get it merged.
> > +1. Run `make tag` to tag the new release and push it to GitHub.
> > +1. Head to https://github.com/prometheus/prometheus/releases and create a new release for the pushed tag.
> > +1. Attach the relevant changelog excerpt to the release on GitHub.
> > +1. Build `linux-amd64`, `linux-386`, and `darwin-amd64` binaries via `make tarball` and attach them to the release.
> > +1. Update the `stable` branch to point to the new release (to tag the latest release image on https://registry.hub.docker.com/u/prom/prometheus/):
> > +
> > +   ```bash
> > +   git branch -f stable <new-release-tag>
> > +   git push origin stable
> > +   ```
> > +
> > +1. Announce the release on the mailing list and on Twitter.
>
> We also need to communicate how to handle doc. Last time I checked
> documentation of many unreleased features went into master directly.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/prometheus/prometheus/pull/930/files#r35524793>.
>